src/dump.c

```diff
@@ -1714,15 +1686,16 @@ static jl_value_t *jl_deserialize_value_(jl_serializer_state *s, jl_value_t *vta
     else if (vtag == (jl_value_t*)jl_datatype_type || vtag == (jl_value_t*)SmallDataType_tag) {
         int32_t sz = (vtag == (jl_value_t*)SmallDataType_tag ? read_uint8(s->s) : read_int32(s->s));
         jl_value_t *v = jl_gc_alloc(ptls, sz, NULL);
+        jl_set_typeof(v, (void*)(intptr_t)0x40);
```

Comment:
no, it's just a random garbage value (0x4, since it's the 4th one added here), like setting it to NULL, but more likely to fail